pliers/extractors/models.py

```diff
@@ -1,67 +1,106 @@
 ''' Extractor classes based on pre-trained models. '''
 
 import numpy as np
-from PIL import Image
 from pliers.extractors.image import ImageExtractor
 from pliers.extractors.base import ExtractorResult
+from pliers.filters.image import ImageResizingFilter
 from pliers.utils import attempt_to_import, verify_dependencies
 
 
 tf = attempt_to_import('tensorflow')
+attempt_to_import('tensorflow.keras')
 
 
-def _resize_image(image, shape):
-    return np.array(
-        Image.fromarray(image).resize(shape, resample=Image.BICUBIC))
-
-
-class TensorFlowKerasInceptionV3Extractor(ImageExtractor):
+class TensorFlowKerasApplicationExtractor(ImageExtractor):
 
     ''' Labels objects in images using a pretrained Inception V3 architecture
     implemented in TensorFlow / Keras.
 
-    Images must be RGB and have shape (299, 299). Images will be resized (with
-    some distortion) if the shape is different.
+    Images must be RGB and be a certain shape. Different model architectures
+    may require different shapes, and images will be resized (with some
+    distortion) if the shape of the image is different.
 
     Args:
+        architecture (str): model architecture to use. One of 'vgg16', 'vgg19',
+            'resnet50', 'inception_resnetv2', 'inceptionv3', 'xception',
+            'densenet121', 'densenet169', 'nasnetlarge', or 'nasnetmobile'.
         weights (str): URL to download pre-trained weights. If None (default),
-            uses the pre-trained Inception V3 model (dated 2017-03-10) used in
-            Keras Applications.
+            uses the pre-trained weights trained on ImageNet used in Keras
+            Applications.
         num_predictions (int): Number of top predicted labels to retain for
             each image.
      '''
 
-    _log_attributes = ('weights', 'num_predictions')
+    _log_attributes = ('architecture', 'weights', 'num_predictions')
     VERSION = '1.0'
 
-    def __init__(self, weights=None, num_predictions=5):
+    def __init__(self,
+                 architecture='inceptionv3',
+                 weights=None,
+                 num_predictions=5):
         verify_dependencies(['tensorflow'])
-        super(TensorFlowKerasInceptionV3Extractor, self).__init__()
+        verify_dependencies(['tensorflow.keras'])
+        super(TensorFlowKerasApplicationExtractor, self).__init__()
+
+        # Model name: (model module, model function, required shape).
```

Comment:
Consider using a NamedTuple or some structure to avoid needing hard-coded indexing later. Not a big deal though.